### PR TITLE
Implement worker origin member of WorkerLocation

### LINKED ISSUE #23331: Implement worker origin member of WorkerLocation
Spec: https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-origin
Code: `components/script/dom/webidls/WorkerLocation.webidl`, `components/script/dom/workerlocation.rs`
Tests: `./mach test-wpt tests/wpt/web-platform-tests/workers/`

You can use `Location::origin` from `components/script/dom/location.rs` as a model for this.

### PR DESCRIPTION
**Pull request by csmoe**
Closes #23331 
r?@jdm 
(cannot build servo on my mac pro, hope CI helps)

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23335)

### CODE REVIEW

File: components/script/dom/workerlocation.rs [hunk 1]
```diff
@@ -34,6 +34,11 @@ impl WorkerLocation {
             WorkerLocationBinding::Wrap,
         )
     }
+
+    // https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-origin
+    pub fn origin(&self) -> &MutableOrigin {
```
Comment (CYBAI) [inline components/script/dom/workerlocation.rs]: This module didn't import `MutableOrigin` but you use it here.
So, I believe you'll want to import it.

File: components/script/dom/workerlocation.rs [hunk 2]
```diff
@@ -34,6 +34,11 @@ impl WorkerLocation {
             WorkerLocationBinding::Wrap,
         )
     }
+
+    // https://html.spec.whatwg.org/multipage/workers.html#dom-workerlocation-origin
```
Comment (CYBAI) [inline components/script/dom/workerlocation.rs]: Tidy complains about it 👀

```
./components/script/dom/workerlocation.rs: to WHATWG may break in the future, use this format instead: https://html.spec.whatwg.org/multipage/#dom-workerlocation-origin
```

**Review by CYBAI — CHANGES_REQUESTED** [review]

File: components/script/dom/workerlocation.rs (hunk 1 above)
Comment (csmoe) [reply components/script/dom/workerlocation.rs]: oops, thanks :)

**Review by jdm — APPROVED** [review]

**Comment by jdm** [comment]
Between https://build.servo.org/builders/linux-rel-wpt/builds/12340/steps/shell__4/logs/filtered-wpt-errorsummary.log and https://build.servo.org/builders/linux-rel-css/builds/12282/steps/shell__4/logs/filtered-wpt-errorsummary.log there are a bunch of tests that are now passing :D

**Comment by jdm** [comment]
@csmoe Are you going to finish this work by updating the expected test results?

**Comment by csmoe** [comment]
@jdm sorry, I have no idea why the ci was unhappy, any tips for that?

**Comment by CYBAI** [comment]
@csmoe As @jdm mentioned at https://github.com/servo/servo/pull/23335#issuecomment-493580515 , there're a bunch of tests that are passing now! but these old builds are deleted, you cannot check the result now. So, @jdm helped to `try=wpt` again. You can check the new logs at https://build.servo.org/builders/linux-rel-wpt/builds/12428/steps/shell__4/logs/filtered-wpt-errorsummary.log and https://build.servo.org/builders/linux-rel-css/builds/12370/steps/shell__4/logs/filtered-wpt-errorsummary.log .

In short, just need to update the test expectations for them! :)

**Comment by jdm** [comment]
https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations describes how to update the tests all at once - you can download https://build.servo.org/builders/linux-rel-wpt/builds/12428/steps/test__1/logs/test-wpt.log/text and https://build.servo.org/builders/linux-rel-css/builds/12370/steps/test/logs/test-wpt.log/text and use those as the input log files.

**Comment by csmoe** [comment]
@CYBAI @jdm really thank, will address this soon.
(git warned about some merge conflicts from tests, so I pushed --force for another rebuilding to get the tests log)

**Comment by jdm** [comment]
https://build.servo.org/builders/linux-rel-css/builds/12377/steps/test/logs/test-wpt.log/text and https://build.servo.org/builders/linux-rel-wpt/builds/12435/steps/test__1/logs/test-wpt.log/text

**Comment by jdm** [comment]
Sorry. If you rebase again I'll merge this immediately.

**Comment by KiChjang** [comment]
@csmoe Hmm... This PR still contains merge conflicts that need to be addressed. Mind fixing it?

**Comment by jdm** [comment]
Bleah. I keep not being notified when this pull request is updated. Sorry :( Please leave a comment next time you push!

**Comment by csmoe** [comment]
@jdm request for a `try=wpt`.

**Comment by csmoe** [comment]
> broken_heart Test failed - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/12414)

downloaded the logs and update-wpt, but no diffs or new tests file generated. (:strange:)
anyway, rebased and pushed -f

**Comment by CYBAI** [comment]
@csmoe Did you download https://build.servo.org/builders/linux-rel-wpt/builds/12472/steps/test__1/logs/test-wpt.log/text and https://build.servo.org/builders/linux-rel-css/builds/12414/steps/test/logs/test-wpt.log/text ?

**Comment by csmoe** [comment]
@CYBAI yes, I downloaded them and update-wpt, but nothing happened.

**Comment by CYBAI** [comment]
I wonder I pointed to the right logs but looks like the updated expectation is different with the filtered result 😨

**Comment by jdm** [comment]
Four more passing tests.

**Comment by csmoe** [comment]
@CYBAI request for a `try=wpt`

**Comment by csmoe** [comment]
@jdm seems not my fault?
https://build.servo.org/builders/linux-rel-nogate/builds/5371/steps/shell__5/logs/stdio
